src/DynamoUtilities/PathHelper.cs

```diff
@@ -88,5 +90,64 @@ public static bool HasWritePermissionOnDir(string folderPath)
 
             return writeAllow && !writeDeny;
         }
+
+        /// <summary>
+        /// This is a utility method for checking if given path contains valid XML document.
+        /// </summary>
+        /// <param name="path">path to the target xml file</param>
+        /// <param name="xmlDoc">System.Xml.XmlDocument repensentation of target xml file</param>
+        /// <returns></returns>
+        public static bool isValidXML(string path, out XmlDocument xmlDoc)
+        {
+            // Based on https://msdn.microsoft.com/en-us/library/875kz807(v=vs.110).aspx
+            // Exception thrown indicate invalid XML document path 
+            try
+            {
+                xmlDoc = new XmlDocument();
+                xmlDoc.Load(path);
+                return true;
+            }
+            catch
+            {
+                xmlDoc = null;
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// This is a utility method for checking if given path contains valid Json document.
+        /// </summary>
+        /// <param name="path">path to the target json file</param>
+        /// <param name="fileContents"> string contents of target json file</param>
+        /// <returns></returns>
+        public static bool isValidJson(string path, out string fileContents)
+        {
+            fileContents = File.ReadAllText(path);
```

Comment:
so I'm a little worried about this - consider the case where the file is 25 megs of revit binding info and we keep reading it from disk over and over again to check that it's json... maybe we can think of something better.

Reply:
I had this concern as well, I thought about adding a workspace/workspaceinfo type property internally so that we can remember somehow but does not seem ideal and will always be Json after save.